emcc.py

```diff
@@ -2592,6 +2596,8 @@ def check_bad_eq(arg):
     # Record USE_PTHREADS setting because it controls whether --shared-memory is passed to lld
     elif newargs[i] == '-pthread':
       settings_changes.append('USE_PTHREADS=1')
+    elif newargs[i] in ('-fno-diagnostics-color', '-fdiagnostics-color=never'):
+      options.color = False
```

Comment:
Why not just do `colored_logger.disable()` here and avoid adding options.color member?